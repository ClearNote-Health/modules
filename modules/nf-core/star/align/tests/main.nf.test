nextflow_process {

    name "Test Process STAR_ALIGN"
    script "../main.nf"
    process "STAR_ALIGN"
    tag "modules"
    tag "modules_nfcore"
    tag "star"
    tag "star/align"
    tag "star/genomegenerate"

    setup {
        run("STAR_GENOMEGENERATE") {
            script "../../../star/genomegenerate/main.nf"
            process {
                """
                input[0] = Channel.of([
                    [ id:'test_fasta' ],
                    [ file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta', checkIfExists: true) ]
                ])
                input[1] = Channel.of([
                    [ id:'test_fai' ],
                    [ file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta.fai', checkIfExists: true) ]
                ])
                input[2] = Channel.of([
                    [ id:'test_gtf' ],
                    [ file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.gtf', checkIfExists: true) ]
                ])
                """
            }
        }
    }

    test("homo_sapiens - single_end") {
        config "./nextflow.config"

        when {
            process {
                """
                input[0] = Channel.of([
                    [ id:'test', single_end:true ], // meta map
                    [ file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/fastq/test_rnaseq_1.fastq.gz', checkIfExists: true) ]
                ])
                input[1] = STAR_GENOMEGENERATE.out.index
                input[2] = Channel.of([
                    [ id:'test_gtf' ],
                    [ file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.gtf', checkIfExists: true) ]
                ])
                input[3] = false
                input[4] = 'illumina'
                input[5] = false
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    file(process.out.log_final[0][1]).name,
                    file(process.out.log_out[0][1]).name,
                    file(process.out.bam[0][1]).name,
                    file(process.out.bam_sorted[0][1]).name,
                    process.out.bam_transcript,
                    process.out.bam_unsorted,
                    process.out.bedgraph,
                    process.out.fastq,
                    process.out.junction,
                    process.out.log_progress,
                    process.out.read_per_gene_tab,
                    process.out.sam,
                    process.out.spl_junc_tab,
                    process.out.tab,
                    process.out.wig,
                    process.out.versions
                ).match("single_end")}
            )
        }
    }

    test("homo_sapiens - paired_end") {
        config "./nextflow.config"

        when {
            process {
                """
                input[0] = Channel.of([
                    [ id:'test', single_end:false ], // meta map
                    [
                        file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/fastq/test_rnaseq_1.fastq.gz', checkIfExists: true),
                        file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/fastq/test_rnaseq_2.fastq.gz', checkIfExists: true)
                    ]
                ])
                input[1] = STAR_GENOMEGENERATE.out.index
                input[2] = Channel.of([
                    [ id:'test_gtf' ],
                    [ file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.gtf', checkIfExists: true) ]
                ])
                input[3] = false
                input[4] = 'illumina'
                input[5] = false
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    file(process.out.log_final[0][1]).name,
                    file(process.out.log_out[0][1]).name,
                    file(process.out.bam[0][1]).name,
                    file(process.out.bam_sorted[0][1]).name,
                    process.out.bam_transcript,
                    process.out.bam_unsorted,
                    process.out.bedgraph,
                    process.out.fastq,
                    process.out.junction,
                    process.out.log_progress,
                    process.out.read_per_gene_tab,
                    process.out.sam,
                    process.out.spl_junc_tab,
                    process.out.tab,
                    process.out.wig,
                    process.out.versions
                ).match("paired_end")}
            )
        }
    }

    test("homo_sapiens - paired_end - arriba") {
        config "./nextflow.arriba.config"

        when {
            process {
                """
                input[0] = Channel.of([
                    [ id:'test', single_end:false ], // meta map
                    [
                        file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/fastq/test_rnaseq_1.fastq.gz', checkIfExists: true),
                        file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/fastq/test_rnaseq_2.fastq.gz', checkIfExists: true)
                    ]
                ])
                input[1] = STAR_GENOMEGENERATE.out.index
                input[2] = Channel.of([
                    [ id:'test_gtf' ],
                    [ file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.gtf', checkIfExists: true) ]
                ])
                input[3] = false
                input[4] = 'illumina'
                input[5] = false
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    file(process.out.log_final[0][1]).name,
                    file(process.out.log_out[0][1]).name,
                    file(process.out.bam[0][1]).name,
                    process.out.bam_sorted,
                    process.out.bam_transcript,
                    process.out.bam_unsorted,
                    process.out.bedgraph,
                    process.out.fastq,
                    process.out.junction,
                    process.out.log_progress,
                    process.out.read_per_gene_tab,
                    process.out.sam,
                    process.out.spl_junc_tab,
                    process.out.tab,
                    process.out.wig,
                    process.out.versions
                ).match("arriba")}
            )
        }
    }

    test("homo_sapiens - paired_end - starfusion") {
        config "./nextflow.starfusion.config"

        when {
            process {
                """
                input[0] = Channel.of([
                    [ id:'test', single_end:false ], // meta map
                    [
                        file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/fastq/test_rnaseq_1.fastq.gz', checkIfExists: true),
                        file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/fastq/test_rnaseq_2.fastq.gz', checkIfExists: true)
                    ]
                ])
                input[1] = STAR_GENOMEGENERATE.out.index
                input[2] = Channel.of([
                    [ id:'test_gtf' ],
                    [ file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.gtf', checkIfExists: true) ]
                ])
                input[3] = false
                input[4] = 'illumina'
                input[5] = false
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    file(process.out.log_final[0][1]).name,
                    file(process.out.log_out[0][1]).name,
                    file(process.out.bam[0][1]).name,
                    process.out.bam_sorted,
                    process.out.bam_transcript,
                    process.out.bam_unsorted,
                    process.out.bedgraph,
                    process.out.fastq,
                    file(process.out.junction[0][1]).name,
                    process.out.log_progress,
                    process.out.read_per_gene_tab,
                    process.out.sam,
                    process.out.spl_junc_tab,
                    process.out.tab,
                    process.out.wig,
                    process.out.versions
                ).match("starfusion")}
            )
        }
    }

    test("homo_sapiens - paired_end - multiple") {
        config "./nextflow.config"

        when {
            process {
                """
                input[0] = Channel.of([
                    [ id:'test', single_end:false ], // meta map
                    [
                        file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/fastq/test_rnaseq_1.fastq.gz', checkIfExists: true),
                        file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/fastq/test_rnaseq_2.fastq.gz', checkIfExists: true),
                        file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/fastq/test_rnaseq_1.fastq.gz', checkIfExists: true),
                        file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/fastq/test_rnaseq_2.fastq.gz', checkIfExists: true)
                    ]
                ])
                input[1] = STAR_GENOMEGENERATE.out.index
                input[2] = Channel.of([
                    [ id:'test_gtf' ],
                    [ file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.gtf', checkIfExists: true) ]
                ])
                input[3] = false
                input[4] = 'illumina'
                input[5] = false
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    file(process.out.log_final[0][1]).name,
                    file(process.out.log_out[0][1]).name,
                    file(process.out.bam[0][1]).name,
                    process.out.bam_sorted,
                    process.out.bam_transcript,
                    process.out.bam_unsorted,
                    process.out.bedgraph,
                    process.out.fastq,
                    process.out.junction,
                    process.out.log_progress,
                    process.out.read_per_gene_tab,
                    process.out.sam,
                    process.out.spl_junc_tab,
                    process.out.tab,
                    process.out.wig,
                    process.out.versions
                ).match("multiple")}
            )
        }
    }

    test("homo_sapiens - single_end - stub") {
        config "./nextflow.config"
        options "-stub"

        when {
            process {
                """
                input[0] = Channel.of([
                    [ id:'test', single_end:true ], // meta map
                    [ file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/fastq/test_rnaseq_1.fastq.gz', checkIfExists: true) ]
                ])
                input[1] = STAR_GENOMEGENERATE.out.index
                input[2] = Channel.of([
                    [ id:'test_gtf' ],
                    [ file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.gtf', checkIfExists: true) ]
                ])
                input[3] = false
                input[4] = 'illumina'
                input[5] = false
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match("stub") }
            )
        }
    }
}