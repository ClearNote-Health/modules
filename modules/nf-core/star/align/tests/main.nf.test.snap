{
    "stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.Log.final.out:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.Log.out:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "10": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.SJ.out.tab:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "11": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.ReadsPerGene.out.tab:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "12": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.Chimeric.out.junction:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "13": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.out.sam:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "14": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.Signal.UniqueMultiple.str1.out.wig:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "15": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.Signal.UniqueMultiple.str1.out.bg:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.Log.progress.out:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "3": [
                    "versions.yml:md5,cfd98c93d4fed5e98dc7f54288ca30fd"
                ],
                "4": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        [
                            "test.Aligned.sortedByCoord.out.bam:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "test.sortedByCoord.out.bam:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "testXd.out.bam:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ]
                    ]
                ],
                "5": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        [
                            "test.Aligned.sortedByCoord.out.bam:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "test.sortedByCoord.out.bam:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ]
                    ]
                ],
                "6": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.toTranscriptome.out.bam:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "7": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.Aligned.unsort.out.bam:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "8": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        [
                            "test.unmapped_1.fastq.gz:md5,68b329da9893e34099c7d8ad5cb9c940",
                            "test.unmapped_2.fastq.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                        ]
                    ]
                ],
                "9": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        [
                            "test.ReadsPerGene.out.tab:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "test.SJ.out.tab:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "test.tab:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ]
                    ]
                ],
                "bam": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        [
                            "test.Aligned.sortedByCoord.out.bam:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "test.sortedByCoord.out.bam:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "testXd.out.bam:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ]
                    ]
                ],
                "bam_sorted": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        [
                            "test.Aligned.sortedByCoord.out.bam:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "test.sortedByCoord.out.bam:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ]
                    ]
                ],
                "bam_transcript": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.toTranscriptome.out.bam:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "bam_unsorted": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.Aligned.unsort.out.bam:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "bedgraph": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.Signal.UniqueMultiple.str1.out.bg:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "fastq": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        [
                            "test.unmapped_1.fastq.gz:md5,68b329da9893e34099c7d8ad5cb9c940",
                            "test.unmapped_2.fastq.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                        ]
                    ]
                ],
                "junction": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.Chimeric.out.junction:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "log_final": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.Log.final.out:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "log_out": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.Log.out:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "log_progress": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.Log.progress.out:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "read_per_gene_tab": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.ReadsPerGene.out.tab:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "sam": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.out.sam:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "spl_junc_tab": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.SJ.out.tab:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "tab": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        [
                            "test.ReadsPerGene.out.tab:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "test.SJ.out.tab:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "test.tab:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,cfd98c93d4fed5e98dc7f54288ca30fd"
                ],
                "wig": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.Signal.UniqueMultiple.str1.out.wig:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.1"
        },
        "timestamp": "2024-05-23T14:29:41.596856274"
    },
    "single_end": {
        "content": [
            "test.Log.final.out",
            "test.Log.out",
            "test.Aligned.sortedByCoord.out.bam",
            "test.Aligned.sortedByCoord.out.bam",
            [
                
            ],
            [
                
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": true
                    },
                    [
                        "test.Signal.Unique.str1.out.bg:md5,c56fc1472776fb927eaf62d973da5f9a",
                        "test.Signal.UniqueMultiple.str1.out.bg:md5,e93373cf6f2a2a9506e2efdb260cdd4f"
                    ]
                ]
            ],
            [
                
            ],
            [
                
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": true
                    },
                    "test.Log.progress.out:md5,b2bd061d6cbaaf3d6d3b1fed547f69b8"
                ]
            ],
            [
                
            ],
            [
                
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": true
                    },
                    "test.SJ.out.tab:md5,75a516ab950fb958f40b29996474949c"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": true
                    },
                    "test.SJ.out.tab:md5,75a516ab950fb958f40b29996474949c"
                ]
            ],
            [
                
            ],
            [
                "versions.yml:md5,cfd98c93d4fed5e98dc7f54288ca30fd"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.1"
        },
        "timestamp": "2024-05-27T08:54:43.290846044"
    },
    "starfusion": {
        "content": [
            "test.Log.final.out",
            "test.Log.out",
            "test.Aligned.out.bam",
            [
                
            ],
            [
                
            ],
            [
                
            ],
            [
                
            ],
            [
                
            ],
            "test.Chimeric.out.junction",
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.Log.progress.out:md5,a835754f030ff0e3e1770842933448d3"
                ]
            ],
            [
                
            ],
            [
                
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.SJ.out.tab:md5,19c3faa1bfa9a0cc5e4c45f17065b53a"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.SJ.out.tab:md5,19c3faa1bfa9a0cc5e4c45f17065b53a"
                ]
            ],
            [
                
            ],
            [
                "versions.yml:md5,cfd98c93d4fed5e98dc7f54288ca30fd"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.1"
        },
        "timestamp": "2024-05-27T09:18:29.353006951"
    },
    "arriba": {
        "content": [
            "test.Log.final.out",
            "test.Log.out",
            "test.Aligned.out.bam",
            [
                
            ],
            [
                
            ],
            [
                
            ],
            [
                
            ],
            [
                
            ],
            [
                
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.Log.progress.out:md5,a42cfda83122bcda0dc51de79270d5eb"
                ]
            ],
            [
                
            ],
            [
                
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.SJ.out.tab:md5,5155c9fd1f787ad6d7d80987fb06219c"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.SJ.out.tab:md5,5155c9fd1f787ad6d7d80987fb06219c"
                ]
            ],
            [
                
            ],
            [
                "versions.yml:md5,cfd98c93d4fed5e98dc7f54288ca30fd"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.1"
        },
        "timestamp": "2024-05-27T09:14:40.709553627"
    },
    "multiple": {
        "content": [
            "test.Log.final.out",
            "test.Log.out",
            "test.Aligned.sortedByCoord.out.bam",
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.Aligned.sortedByCoord.out.bam:md5,82806aebd8d56094aa66c1111a7b15f6"
                ]
            ],
            [
                
            ],
            [
                
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    [
                        "test.Signal.Unique.str1.out.bg:md5,d7bf8b70b436ca048a62513e1d0ece3a",
                        "test.Signal.UniqueMultiple.str1.out.bg:md5,686d58493b9eb445b56ace4d67f76ef6"
                    ]
                ]
            ],
            [
                
            ],
            [
                
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.Log.progress.out:md5,8007a94c531af26fe9c034aa300a8b19"
                ]
            ],
            [
                
            ],
            [
                
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.SJ.out.tab:md5,069877e053714e23010fe4e1c003b4a2"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.SJ.out.tab:md5,069877e053714e23010fe4e1c003b4a2"
                ]
            ],
            [
                
            ],
            [
                "versions.yml:md5,cfd98c93d4fed5e98dc7f54288ca30fd"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.1"
        },
        "timestamp": "2024-05-27T09:19:55.034986098"
    },
    "paired_end": {
        "content": [
            "test.Log.final.out",
            "test.Log.out",
            "test.Aligned.sortedByCoord.out.bam",
            "test.Aligned.sortedByCoord.out.bam",
            [
                
            ],
            [
                
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    [
                        "test.Signal.Unique.str1.out.bg:md5,d7bf8b70b436ca048a62513e1d0ece3a",
                        "test.Signal.UniqueMultiple.str1.out.bg:md5,686d58493b9eb445b56ace4d67f76ef6"
                    ]
                ]
            ],
            [
                
            ],
            [
                
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.Log.progress.out:md5,b2bd061d6cbaaf3d6d3b1fed547f69b8"
                ]
            ],
            [
                
            ],
            [
                
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.SJ.out.tab:md5,844af19ab0fc8cd9a3f75228445aca0d"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.SJ.out.tab:md5,844af19ab0fc8cd9a3f75228445aca0d"
                ]
            ],
            [
                
            ],
            [
                "versions.yml:md5,cfd98c93d4fed5e98dc7f54288ca30fd"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.1"
        },
        "timestamp": "2024-05-27T08:55:34.198079567"
    }
}